### PR TITLE
Kafkasource v1beta1 reconciler

### LINKED ISSUE #1395: Make reconcilers use KafkaSource and KafkaBinding v1beta1 resources
Part of https://github.com/knative/eventing-contrib/issues/1332

Make reconcilers use v1beta1 resources

### PR DESCRIPTION
**Pull request by aliok**
Fixes #1395

## Proposed Changes

- Use KafkaSource and KafkaBinding v1beta1 in reconciler
- Change condition `KeyResourcesCorrect` to `KeyTypeCorrect` as it resources don't exist in spec anymore and key type lifecycle checks was using `KeyResourcesCorrect` mistakenly

**Release Note**

```release-note
- 🧽 Update or clean up current behavior
Reconcile KafkaSource and KafkaBinding using v1beta1 api shape. Operate on dependent resources (Subscriptions, etc.) using their v1 shapes. Note that `resource` and `serviceAccountName` fields are removed from the types in v1beta1.
```

**Docs**

### CODE REVIEW

File: kafka/source/pkg/reconciler/source/controller.go [hunk 1]
```diff
@@ -33,8 +33,8 @@ import (
 	"knative.dev/pkg/resolver"
 
 	kafkaclient "knative.dev/eventing-contrib/kafka/source/pkg/client/injection/client"
-	kafkainformer "knative.dev/eventing-contrib/kafka/source/pkg/client/injection/informers/sources/v1alpha1/kafkasource"
-	"knative.dev/eventing-contrib/kafka/source/pkg/client/injection/reconciler/sources/v1alpha1/kafkasource"
+	kafkainformer "knative.dev/eventing-contrib/kafka/source/pkg/client/injection/informers/sources/v1beta1/kafkasource"
+	"knative.dev/eventing-contrib/kafka/source/pkg/client/injection/reconciler/sources/v1beta1/kafkasource"
 )
 
 func NewController(
```
Comment (aliok) [inline kafka/source/pkg/reconciler/source/controller.go:40]: Can't use "knative.dev/eventing/pkg/apis/sources/v1alpha1" yet as it is not released yet.

Comment (aliok) [reply kafka/source/pkg/reconciler/source/controller.go:40]: Not sure if that's a blocker for this PR

Comment (matzew) [reply kafka/source/pkg/reconciler/source/controller.go:40]: perhaps go w/ "latest" of the dependency ? (aka master branch)

Comment (aliok) [reply kafka/source/pkg/reconciler/source/controller.go:40]: I need to learn more about Go modules :D

**Comment by aliok** [comment]
ready for review
cc @matzew @lionelvillard @vaikas 

About the comment in https://github.com/knative/eventing-contrib/pull/1405#discussion_r459474431 : let's do that part later

File: kafka/source/pkg/reconciler/source/controller.go (hunk 1 above)
Comment (aliok) [reply kafka/source/pkg/reconciler/source/controller.go:40]: Let me do this later. Nothing unique from that package is used anyway. Only a `Kind` function which is identical in v1alpha1.
When the dependency to eventing is changed to master I can do it.

**Comment by vaikas** [comment]
/lgtm
/hold
for others to chime in. Also, might want to mention that the resource spec does not exist in the release notes?

**Comment by aliok** [comment]
> Also, might want to mention that the resource spec does not exist in the release notes?

Updated the PR desc. all good

**Pull request merged**